src/embed/connection.rs

```diff
@@ -273,15 +273,19 @@ async fn do_put(txn: &RwLock<Transaction<'_>>, req: PutRequest) -> Result<PutRes
     Ok(PutResponse {})
 }
 
-// This fn just forwards to sync::begin_sync at this point. execute() requires its
-// do_ers to take txns even if they don't need it so this fn also helps begin_sync
-// have a simpler signature.
 async fn do_begin_sync<'a, 'b>(
     _: &'a dag::Store,
     _: &'b TxnMap<'a>,
     req: BeginSyncRequest,
 ) -> Result<BeginSyncResponse, sync::BeginSyncError> {
-    let begin_sync_response = sync::begin_sync(&req).await?;
+    // TODO move client up to process() or into a lazy static so we can re-use.
+    let hyper_client_owned: hyper::Client<hyper::client::HttpConnector>;
+    let mut hyper_client_option = None;
+    if !sync::USE_BROWSER_FETCH {
+        hyper_client_owned = hyper::Client::new();
+        hyper_client_option = Some(&hyper_client_owned);
+    }
+    let begin_sync_response = sync::begin_sync(hyper_client_option, &req).await?;
```

Comment:
This is kind of cool -- I presume that http client is going to get passed all over the place to async functionality and it's totally fine because it's immutable.

Reply:
yes it is seeming like inner mutability is an important thing for shared state